### DISCUSSION TITLE
Need a way to determine if an swf file is built in debug or release mode

### DISCUSSION THREAD

**Discussion by ylazy** [airsdk/Adobe-Runtime-Support#1572] [General]
Bellow is the only way to determine if an swf file is built in debug mode or not:
```
var stackTrace:String = new Error().getStackTrace();
var isDebugBuild:Boolean = stackTrace && stackTrace.search(/:[0-9]+]$/m) != -1; // stackTrace contains :lineNumber in debug build.
```
see https://stackoverflow.com/a/1091872

But that is only possible if the swf file is running on debug runtime, for example on AIR Debug Launcher.
I can't find a way to detect if an swf file is built in debug mode or release mode, on Adobe AIR Runtime (release runtime).

Please help!
Thanks!

**Comment by ajwfrost** [comment]
Hi

I don't think there is a way to do this, other than to have a property set up in your main app class that is built with a variable set via a 'config' block of ActionScript that you change depending on the build flavour.

The one thing we might be able to do is to look for "debugfile" ActionScript instructions and set a flag if any of these are found. It could slow things down a little but if we limit it to just "debugfile" (rather than "debugline" and "debug" instructions which are more frequent) then hopefully it wouldn't be too much of an impact - plus it would only be an impact if you did actually have these instructions.

I don't know that we'd be able to check what file these came from though, i.e. if you'd linked in SWCs or loaded in other SWFs at runtime, these would be inspected as well but at this level in the AVM+ engine (where we would see the existence of debugfile etc) we wouldn't have the SWF filenames. So we could just add a flag like `System.containsDebugInfo` that would be set if the virtual machine has ever encountered one of these debugfile instructions... would that be useful?

thanks

**Reply by ylazy** [reply]
> So we could just add a flag like System.containsDebugInfo that would be set if the virtual machine has ever encountered one of these debugfile instructions... would that be useful?

Yes, it would be very useful and I think it's exactly what I need.

Thank you!

**Reply by ajwfrost** [reply]
Fairly simple to do :-) it will be in our next major release of AIR....
